src/openforms/submissions/tokens.py

```diff
@@ -95,13 +26,31 @@ def _make_hash_value(
             str(getattr(submission_report, attribute) or "")
             for attribute in submission_report_attributes
         ]
-        return "".join(submission_report_bits) + str(timestamp)
+        return submission_report_bits
+
+
+class SubmissionStatusTokenGenerator(BaseTokenGenerator):
+    key_salt = "openforms.submissions.tokens.SubmissionStatusTokenGenerator"
+    # pin to the minimum of 1 day - the SDK should not be polling for longer than
+    # about a minute anyway.
+    token_timeout_days = 1
 
-    def _num_days(self, dt) -> int:
+    def get_hash_value_parts(self, submission: Submission) -> List[str]:
         """
-        Return the number of days between 01-01-2001 and today
+        Obtain the attribute values that mutate to invalidate the token.
+
+        Technically this token is not single-use, as the status endpoint using this
+        token is polled until processing has completed. However, if the on-completation
+        chain was re-started (for example), the token is invalidated since it no longer
+        represents the current state of execution.
         """
-        return (dt - date(2001, 1, 1)).days
+        attributes = [
+            "completed_on",
+            "suspended_on",
+            "on_completion_task_id",
+        ]
+        return [str(getattr(submission, attr)) for attr in attributes]
```

Comment:
I think that `suspended_on` could be `None`? In this case one of the attributes returned would be `"None"`. If this is the case, it may be good to use the same pattern as in the submission_report, where it is:
`str(getattr(submission_report, attribute) or "")`

Reply:
It can be indeed, but I don't think that's a problem as a "legal" value is completely different from the string `"None"` and it's only used as input to the hashing function, so I don't consider it a problem.